src/modalities/api.py

```diff
@@ -95,13 +116,37 @@ def convert_pytorch_to_hf_checkpoint(
     return hf_model
 
 
+def shuffle_tokenized_data(
+    input_data_path: Path,
+    output_data_path: Path,
+    batch_size: int,
+    file_existence_policy: FileExistencePolicy,
+    seed: int = None,
+):
+    """Shuffles a tokenized file (.pbin) and stores it on disc.
+
+    Args:
+        input_data_path (Path): File path to the tokenized data (.pbin).
+        output_data_path (Path): File path to write the shuffled tokenized data.
+        batch_size (int): Number of documents to process per batch.
+        file_existence_policy (FileExistencePolicy): Policy to apply when the output file already exists.
+    """
+    if output_data_path.exists():
+        if not enforce_file_existence_policy(output_data_path, file_existence_policy):
+            return
+
+    TokenizedDataShuffler.shuffle_tokenized_data(
+        input_data_path=input_data_path, output_data_path=output_data_path, batch_size=batch_size, seed=seed
+    )
+
+
 def create_shuffled_dataset_chunk(
     file_path_list: list[Path],
     output_chunk_file_path: Path,
     chunk_id: int,
     num_chunks: int,
-    vocab_size: int,
-    shuffle: bool = True,
+    file_existence_policy: FileExistencePolicy,
+    global_seed: int = None,
```

Comment:
```suggestion
    global_seed: Optional[int] = None,
```
would require `from typing import Optional`.